src/cc/frontends/clang/kbuild_helper.cc

```diff
@@ -56,6 +60,17 @@ int KBuildHelper::get_flags(const char *uname_machine, vector<string> *cflags) {
     arch = "arm64";
   }
 
+  // If ARCH env is defined, use it over uname
+  archenv = getenv("ARCH");
+  if (!archenv)
+    archenv = getenv("BCC_ARCH");
+
+  if (archenv) {
+    if (getenv("BCC_KBUILD_DEBUG"))
```

Comment:
Can we use one of existing debug flags for output here? For example, debug=4 for preprocessor result can be used since this impacts the clang commands?

Reply:
I'll look into that, thanks

Reply:
Sure, I'll reuse `DEBUG_PREPROCESSOR` to print the chose ARCH. Can I also reuse `DEBUG_PREPROCESSOR` to print the kernel source directory path or does another flag make more sense?

Reply:
You can just reuse `DEBUG_PREPROCESSOR` to print the kernel source directoryy path.

Reply:
Could you think of a way to put the code dealing different architectures in one place (header is okay), so the main.c file will be clean?

Reply:
@yonghong-song didn't follow your comment. When you say `main.c`, do you mean the `/virtual/main.c` file?

Also the code dealing with different architectures is already in a separate header (`helpers.h`). Do you mean the code shared between `loader.cc` and `b_frontend_action.cc` should be made common? I think we can do that using function pointers and enums. So we could define one enum per arch, and pass a function pointer callback to generic code that handles different arches, which would call then call the callback after detecting the ARCH. That would keep the "arch detection logic" separate from the code that "uses the arch and does different things".

Reply:
Note that I would keep `helpers.h` as is in the original patch though since that code isn't shared by the CC code (unless I missed something).

Reply:
Sorry for confusion. yes, I do `mean the code shared between loader.cc and b_frontend_action.cc should be made common`.

Reply:
Got it.